pennylane/ops/qubit/qchem_ops.py

```diff
@@ -114,12 +124,39 @@ def compute_matrix(phi):  # pylint: disable=arguments-differ
                 [ 0.0000,  0.2474,  0.9689,  0.0000],
                 [ 0.0000,  0.0000,  0.0000,  1.0000]])
         """
-        c = qml.math.cos(phi / 2)
+        interface = qml.math.get_interface(phi)
+        if interface == "tensorflow":
+            c = qml.math.cos(phi / 2)
+            s = qml.math.sin(phi / 2)
+            ones = qml.math.ones_like(phi)
+            zeros = qml.math.zeros_like(phi)
+            rows = [
+                [ones, zeros, zeros, zeros],
+                [zeros, c, -s, zeros],
+                [zeros, s, c, zeros],
+                [zeros, zeros, zeros, ones],
+            ]
+            return qml.math.stack([stack_last(row) for row in rows], axis=-2)
+
+        c_minus_1 = qml.math.cos(phi / 2) - 1
         s = qml.math.sin(phi / 2)
 
-        mat = qml.math.diag([1, c, c, 1])
-        off_diag = qml.math.convert_like(np.diag([0, 1, -1, 0])[::-1].copy(), phi)
-        return mat + s * qml.math.cast_like(off_diag, s)
```

Comment:
The new addition is much larger than the initial function. I am curious if the gain in performance etc. justifies this big complication?

Reply:
I agree, unfortunately the functions get quite big. This is due to supporting broadcasting (vector `phi` -> 3d matrix returned) while not deprecating the performance in the non-broadcasting case (scalar `phi` -> 2d matrix). `compute_matrix` is called a lot in the PL pipeline, so I think performance should be maintained here.
Happy to hear suggestions that may shorten the functions :)